### ISSUE TITLE
“winget install -l” The specified location is different from the installation location

### ISSUE THREAD

**Issue by XYZliang** [microsoft/winget-cli#2752]
### Brief description of your issue

When I use
`Tencent.TencentDocs -l "E:\Program Files\TencentDocs"`
<img width="815" alt="uTools_1670412866288" src="https://user-images.githubusercontent.com/59312540/206168791-0907be08-bc9d-47ab-ade4-f0bf0bdaf8b2.png">
When I ordered the installation of TencentDocs, the software was incorrectly installed in the directory "E:\Program\" instead of the specified "E:\Program Files\TencentDocs".
<img width="563" alt="image" src="https://user-images.githubusercontent.com/59312540/206169049-2807257d-59d5-497b-aff5-d015847e9581.png">
There is only an empty TencentDocs folder under the E:\Program Files folder
<img width="521" alt="image" src="https://user-images.githubusercontent.com/59312540/206169197-8547a5d4-dabc-4831-802e-1e5c1629e98a.png">
I've tried uninstalling and reinstalling many times, and that seems to happen when I install this particular program. No abnormalities have been found in other installation programs.
I don't know if the problem was on winget or the package, but I decided to raise the issu.

### Steps to reproduce

just run `Tencent.TencentDocs -l "E:\Program Files\TencentDocs"` in cmd

### Expected behavior

TencentDocs will be installed to the directory "E:\Program Files\TencentDocs"

### Actual behavior

Under E:\Program Files, only a new folder TencentDocs was created, but no program was installed and the folder is empty. The Program is installed to E:\Program

### Environment

```shell
Windows 程序包管理器 v1.3.2691
版权所有 (C) Microsoft Corporation。保留所有权利。

Windows: Windows.Desktop v10.0.22621.900
系统体系结构: X64
程序包: Microsoft.DesktopAppInstaller v1.18.2691.0

日志: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

链接
----------------------------------------------------------------------------
隐私声明             https://aka.ms/winget-privacy
许可协议             https://aka.ms/winget-license
第三方声明           https://aka.ms/winget-3rdPartyNotice
主页                 https://aka.ms/winget
Windows 应用商店条款 https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by eabase** [comment]
Yes, what's your Terminal character code settings? Also try to use powershell for better terminal support. Clearly you need to escape the empty space somehow.  What about using single quotes?

**Comment by XYZliang** [comment]
> 是的，您的终端字符代码设置是什么？还可以尝试使用Powershell以获得更好的终端支持。显然，您需要以某种方式逃离空白空间。使用单引号怎么样？

Thank you for your suggestion, I tried to install the software with this problem using powershell, but the problem persists.
Also, both my cmd and powershell encoding is 936, which is the default Simplified Chinese (GB2312). After I switched to UTF-8 (65001) encoding, the problem existed as well.
I also tried using single quotes instead of double quotes, but still no solution. Interestingly, this bug only occurs with some specific software, such as Motrix, Termius, uTools, Tencent Docs.

**Comment by XYZliang** [comment]
> 我想知道这是否与语法有关。字符串可能需要作为嵌套引用以保留空格。

Sorry I don't seem to understand what you mean, can you briefly say how one should try to fix it? I reinstalled my Windows system and tried to install all the software via winget, but there are more and more packages with this problem and it is only those few specific packages that show this exception.

**Comment by denelon** [comment]
From [Microsoft Learn](https://learn.microsoft.com/windows/package-manager/winget/install):

<html>
<body>


-l, --location | Location to install to (if supported).
-- | --



</body>
</html>

It's possible these packages do not support passing the location through to the installer. If they do support this mechanism, then it's likely related to the syntax used to specify the location.

**Comment by XYZliang** [comment]
> 来自[微软学习](https://learn.microsoft.com/windows/package-manager/winget/install)：
> 
> -l， --位置	要安装到的位置（如果支持）。
> 这些包可能不支持将位置传递到安装程序。如果它们确实支持此机制，则它可能与用于指定位置的语法有关。

感谢您的回复，我将这些出现安装路径问题软件安装到别的不带空格的文件夹，例如“D:\ProgramFiles”时，一切正常，所以他们应该支持将位置传递到安装程序。正如你所说，似乎是指定的安装位置中的空格引发了某种语法bug
